Minio/MinioClientConfig.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Minio
+{
+    public class MinioClientConfig
+    {
+        /// <summary>
+        /// Location of the server, supports HTTP and HTTPS
+        /// </summary>
+        public string Endpoint { get; set; }
+
+        /// <summary>
+        /// Access Key for authenticated requests (Optional, can be omitted for anonymous requests)
+        /// </summary>
+        public string AccessKey { get; set; } = "";
+
+        /// <summary>
+        /// Secret Key for authenticated requests (Optional, can be omitted for anonymous requests)
+        /// </summary>
+        public string SecretKey { get; set; } = "";
+
+        /// <summary>
+        /// Optional custom region
+        /// </summary>
+        public string Region { get; set; } = "";
+
+        /// <summary>
+        /// Optional session token
+        /// </summary>
+        public string SessionToken { get; set; } = "";
+
+        /// <summary>
+        /// Change the default for multipart part sizes when uploading through client
+        /// </summary>
+        public long DefaultUploadPartSize { get; set; } = 64 * 1024L * 1024L;
```

Comment:
This is not a client level config, it should be part of PutObject input. That would provide the required flexibility per Upload operation.

Reply:
It is intended to be a client level config, and probably one that doesn't have to be adjusted.    The behavior of `PutObjectAsync` changes when the object size goes above this limit.  Using the minimum part size is simply too small for most cases.

There are already three overloads for `PutObjectAsync`, and to add another parameter to them would result in at least one more.

Reply:
We should add one more in line with our other SDKs such as minio-go https://docs.min.io/docs/golang-client-api-reference.html#PutObject 

It is incorrect to have this configured for MinioClient constructor config, this should be configurable per PutObject() instead.

Reply:
Alright.  I'll look more into what the minio-go client is doing.

Reply:
I've move the extra parameters for `PutObjectAsync` into a new `MinioPutObjectOptions` and moved `DefaultUploadPartSize` option from `MinioClientConfig` to the new `MinioPutObjectOptions`.